hack/scripts/sync-prow.sh

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Check if git-filter-repo tool is installed
+if which git-filter-repo > /dev/null 2>&1; then
+  echo "git-filter-repo is installed"
+else
+  echo "please install the git-filter-repo tool per https://github.com/newren/git-filter-repo"
+  exit 1
+fi
```

Comment:
This isn't really necessary because we already do `set -e` so if the tool isn't on the `$PATH` it will die.

You can use the `if ! command; then ... fi` idiom to only display the error if the command cannot be found or failed for whatever reason. I think it would be simpler to move this there instead.

Reply:
Sorry, I requested this. I'd prefer to know about missing non-default tools before starting git ops.

Would be cleaner if encapsulated in a check_deps() shell function.

Reply:
Moved the checks to `check_deps` function.

Reply:
> Sorry, I requested this. I'd prefer to know about missing non-default tools before starting git ops.

Ah, OK. I would have gone for this option also if this script was going to be used long-term, but as this script is only temporary and there is only 1 niche tool involved I thought dedicating a separate stanza for this felt a bit like overkill. But I'm fine with check_deps() now that we have it.